tests/test_internals/test_roi_extractors_module.py

```diff
@@ -482,6 +482,289 @@ def test_add_two_plane_segmentation(self):
+        )
+
+    def setUp(self):
+        self.segmentation_extractor = generate_dummy_segmentation_extractor(
+            num_rois=self.num_rois,
+            num_frames=self.num_frames,
+            num_rows=self.num_rows,
+            num_columns=self.num_columns,
+        )
+        self.nwbfile = NWBFile(
+            session_description="session_description",
+            identifier="file_id",
+            session_start_time=self.session_start_time,
+        )
+
+        self.metadata = dict(Ophys=dict())
+
+        fluorescence_metadata = dict(
+            Fluorescence=dict(
+                name=self.fluorescence_name,
+                roi_response_series=[
+                    self.raw_roi_response_series_metadata,
+                    self.deconvolved_roi_response_series_metadata,
+                    self.neuropil_roi_response_series_metadata,
+                ],
+            )
+        )
+
+        dff_metadata = dict(
+            DfOverF=dict(
+                name=self.dff_name,
+                roi_response_series=[self.dff_roi_response_series_metadata],
+            )
+        )
+
+        self.metadata["Ophys"].update(fluorescence_metadata)
+        self.metadata["Ophys"].update(dff_metadata)
+
+        self.ophys = get_module(self.nwbfile, "ophys")
+
+    def test_add_fluorescence_traces(self):
+        """Test fluorescence traces are added correctly to the nwbfile."""
+
+        add_fluorescence_traces(
+            segmentation_extractor=self.segmentation_extractor,
+            nwbfile=self.nwbfile,
+            metadata=self.metadata,
+        )
+
+        assert self.fluorescence_name in self.ophys.data_interfaces
+
+        fluorescence = self.ophys.get(self.fluorescence_name)
+
+        self.assertEqual(fluorescence.name, self.fluorescence_name)
+        self.assertEqual(len(fluorescence.roi_response_series), 3)
+
+        self.assertEqual(
+            fluorescence["RoiResponseSeries"].description,
+            self.raw_roi_response_series_metadata["description"],
+        )
+
+        assert self.dff_roi_response_series_metadata["name"] not in fluorescence.roi_response_series
+
+        self.assertEqual(
+            fluorescence["Deconvolved"].description,
+            self.deconvolved_roi_response_series_metadata["description"],
+        )
+
+        self.assertEqual(
+            fluorescence["Neuropil"].unit,
+            self.neuropil_roi_response_series_metadata["unit"],
+        )
+
+        self.assertEqual(
+            fluorescence["Neuropil"].rate,
+            self.segmentation_extractor.get_sampling_frequency(),
+        )
+
+        traces = self.segmentation_extractor.get_traces_dict()
+
+        assert_array_equal(fluorescence["RoiResponseSeries"].data, traces["raw"].T)
+        assert_array_equal(fluorescence["Deconvolved"].data, traces["deconvolved"].T)
+        assert_array_equal(fluorescence["Neuropil"].data, traces["neuropil"].T)
+
+    def test_add_df_over_f_trace(self):
```

Comment:
Given that in your previous test you are checking that the dff is not added to the fluorescence container as a roi response, I wonder what do you think of the following: For this test, having a segmentation extractor with only dff and testing that the fluorescence container is non created in this case.

Reply:
yes, that makes sense.